src/metadata/selectors.js

```diff
@@ -66,6 +66,35 @@ export const getCategoryOptionCombosByCategoryComboId = (
  * selectors that should have a cache per parameter (say an id) should use re-reselect.
  */
 
+export const getCategoryOptionsByCategoryOptionComboId = createCachedSelector(
+    (metadata) => metadata,
+    (_, categoryOptionComboId) => categoryOptionComboId,
+    (metadata, categoryOptionComboId) => {
+        let categoryOptionCombo
+
+        const categoryCombos = Object.values(metadata.categoryCombos)
+
+        // "outer:" is a positional label that can be used to break multiple
+        // loops: https://stackoverflow.com/a/1564838
+        outer: for (const categoryCombo of categoryCombos) {
+            for (const curCategoryOptionCombo of categoryCombo.categoryOptionCombos) {
+                if (curCategoryOptionCombo.id === categoryOptionComboId) {
+                    categoryOptionCombo = curCategoryOptionCombo
+                    break outer
+                }
+            }
+        }
+
+        if (!categoryOptionCombo) {
+            return []
+        }
+
+        return Object.values(metadata.categoryOptions).filter(({ id }) =>
+            categoryOptionCombo.categoryOptions.includes(id)
+        )
+    }
+)((_, categoryOptionComboId) => categoryOptionComboId)
```

Comment:
Wow, this is the first time I've seen the legendary position label used for real. I do think this can be simplied though to avoid it's usage:
```suggestion
export const getCategoryOptionsByCategoryOptionComboId = createCachedSelector(
    (metadata) => metadata,
    (_, categoryOptionComboId) => categoryOptionComboId,
    (metadata, categoryOptionComboId) => {
        const categoryCombos = Object.values(metadata.categoryCombos)

        for (const categoryCombo of categoryCombos) {
            for (const curCategoryOptionCombo of categoryCombo.categoryOptionCombos) {
                if (curCategoryOptionCombo.id === categoryOptionComboId) {
                    return Object.values(metadata.categoryOptions).filter(
                        ({ id }) =>
                            curCategoryOptionCombo.categoryOptions.includes(id)
                    )
                }
            }
        }

        return []
    }
)((_, categoryOptionComboId) => categoryOptionComboId)
```
I ran this locally and the tests still pass. But then I realised there wasn't a test for this one. So this means that:
- I'm not sure this will work as expected (I'm in a lecture and finding it hard to focus)
- We should add a test to verify this complex logic is working as expected

Reply:
I've added a test and implemented your suggestion